--- pvemanagerlib.js.orig	2022-03-17 09:08:40.000000000 -0400
+++ pvemanagerlib.js	2022-04-03 08:54:10.229689187 -0400
@@ -8068,6 +8068,7 @@
     alias: ['widget.pveiScsiProviderSelector'],
     comboItems: [
 	['comstar', 'Comstar'],
+	['truenas', 'TrueNAS-API'],
 	['istgt', 'istgt'],
 	['iet', 'IET'],
 	['LIO', 'LIO'],
@@ -49636,6 +49637,7 @@
 	data: {
 	    isLIO: false,
 	    isComstar: true,
+        isTrueNAS: false,
 	    hasWriteCacheOption: true,
 	},
     },
@@ -49648,10 +49650,26 @@
 	    },
 	},
 	changeISCSIProvider: function(f, newVal, oldVal) {
+	    var me = this;
 	    var vm = this.getViewModel();
 	    vm.set('isLIO', newVal === 'LIO');
 	    vm.set('isComstar', newVal === 'comstar');
-	    vm.set('hasWriteCacheOption', newVal === 'comstar' || newVal === 'istgt');
+	    vm.set('isTrueNAS', newVal === 'truenas');
+	    vm.set('hasWriteCacheOption', newVal === 'comstar' || newVal === 'truenas' || newVal === 'istgt');
+	    if (newVal !== 'truenas') {
+		me.lookupReference('truenas_use_ssl_field').setValue(false);
+		me.lookupReference('truenas_apiv4_host_field').setValue('');
+		me.lookupReference('truenas_user_field').setValue('');
+		me.lookupReference('truenas_user_field').allowBlank = true;
+		me.lookupReference('truenas_password_field').setValue('');
+		me.lookupReference('truenas_password_field').allowBlank = true;
+		me.lookupReference('truenas_confirmpw_field').setValue('');
+		me.lookupReference('truenas_confirmpw_field').allowBlank = true;
+	    } else {
+		me.lookupReference('truenas_user_field').allowBlank = true;
+		me.lookupReference('truenas_password_field').allowBlank = true;
+		me.lookupReference('truenas_confirmpw_field').allowBlank = true;
+	    }
 	},
     },
 
@@ -49669,6 +49687,7 @@
     },
 
     setValues: function(values) {
+        values.truenas_confirmpw = values.truenas_password;
 	values.writecache = values.nowritecache ? 0 : 1;
 	this.callParent([values]);
     },
@@ -49685,7 +49704,7 @@
 		allowBlank: false,
 	    },
 	    {
-		xtype: me.isCreate ? 'textfield' : 'displayfield',
+		xtype: 'textfield',
 		name: 'pool',
 		value: '',
 		fieldLabel: gettext('Pool'),
@@ -49695,11 +49714,11 @@
 		xtype: me.isCreate ? 'textfield' : 'displayfield',
 		name: 'blocksize',
 		value: '4k',
-		fieldLabel: gettext('Block Size'),
+		fieldLabel: gettext('ZFS Block Size'),
 		allowBlank: false,
 	    },
 	    {
-		xtype: me.isCreate ? 'textfield' : 'displayfield',
+		xtype: 'textfield',
 		name: 'target',
 		value: '',
 		fieldLabel: gettext('Target'),
@@ -49710,9 +49729,34 @@
 		name: 'comstar_tg',
 		value: '',
 		fieldLabel: gettext('Target group'),
-		bind: me.isCreate ? { disabled: '{!isComstar}' } : { hidden: '{!isComstar}' },
+		bind: {
+		    hidden: '{!isComstar}'
+		},
 		allowBlank: true,
 	    },
+	    {
+		xtype: 'proxmoxcheckbox',
+		name: 'truenas_use_ssl',
+		reference: 'truenas_use_ssl_field',
+		inputId: 'truenas_use_ssl_field',
+		checked: false,
+		bind: {
+		    hidden: '{!isTrueNAS}'
+		},
+		uncheckedValue: 0,
+		fieldLabel: gettext('API use SSL'),
+	    },
+	    {
+		xtype: 'textfield',
+		name: 'truenas_user',
+		reference: 'truenas_user_field',
+		inputId: 'truenas_user_field',
+		value: '',
+		fieldLabel: gettext('API Username'),
+		bind: {
+		    hidden: '{!isTrueNAS}'
+		},
+	    },
 	];
 
 	me.column2 = [
@@ -49742,7 +49786,9 @@
 		xtype: me.isCreate ? 'textfield' : 'displayfield',
 		name: 'comstar_hg',
 		value: '',
-		bind: me.isCreate ? { disabled: '{!isComstar}' } : { hidden: '{!isComstar}' },
+		bind: {
+		    hidden: '{!isComstar}'
+		},
 		fieldLabel: gettext('Host group'),
 		allowBlank: true,
 	    },
@@ -49750,9 +49796,62 @@
 		xtype: me.isCreate ? 'textfield' : 'displayfield',
 		name: 'lio_tpg',
 		value: '',
-		bind: me.isCreate ? { disabled: '{!isLIO}' } : { hidden: '{!isLIO}' },
-		allowBlank: false,
+		bind: {
+		    hidden: '{!isLIO}'
+		},
 		fieldLabel: gettext('Target portal group'),
+		allowBlank: true
+	    },
+	    {
+		xtype: 'proxmoxtextfield',
+		name: 'truenas_apiv4_host',
+		reference: 'truenas_apiv4_host_field',
+		value: '',
+		editable: true,
+		emptyText: Proxmox.Utils.noneText,
+		bind: {
+		    hidden: '{!isTrueNAS}'
+		},
+		fieldLabel: gettext('API IPv4 Host'),
+	    },
+	    {
+		xtype: 'proxmoxtextfield',
+		name: 'truenas_password',
+		reference: 'truenas_password_field',
+		inputType: me.isCreate ? '' : 'password',
+		value: '',
+		editable: true,
+		emptyText: Proxmox.Utils.noneText,
+		bind: {
+		    hidden: '{!isTrueNAS}'
+		},
+		fieldLabel: gettext('API Password'),
+		change: function(f, value) {
+		    if (f.rendered) {
+			f.up().down('field[name=truenas_confirmpw]').validate();
+		    }
+		},
+	    },
+	    {
+		xtype: 'proxmoxtextfield',
+		name: 'truenas_confirmpw',
+		reference: 'truenas_confirmpw_field',
+		inputType: me.isCreate ? '' : 'password',
+		value: '',
+		editable: true,
+		submitValue: false,
+		emptyText: Proxmox.Utils.noneText,
+		bind: {
+		    hidden: '{!isTrueNAS}'
+		},
+		fieldLabel: gettext('Confirm Password'),
+		validator: function(value) {
+		    var pw = this.up().down('field[name=truenas_password]').getValue();
+		    if (pw !== value) {
+			return "Passwords do not match!";
+		    }
+		    return true;
+		},
 	    },
 	];
 
